collections/deep_merge.ts

```diff
@@ -93,14 +93,14 @@ export function deepMerge<
 }
 
 function mergeObjects(
-  left: NonNullable<object>,
-  right: NonNullable<object>,
+  left: Readonly<NonNullable<unknown>>,
```

Comment:
Why has this been changed to `unknown`?

Reply:
Good catch!
Because It was incompatible with the signature of `deepMerge` (`'object' is not assignable to parameter of type 'Readonly<Record<PropertyKey, unknown>>'`)
The typing issue is now fixed by making `isMergeable` a type guard of `Record<PropertyKey, unknown>` instead. This aligns better with what `isMergeable` is trying to accomplish and match the signature of `deepMerge`.